includes/payment-methods/class-wc-stripe-express-checkout-element.php

```diff
@@ -585,6 +580,34 @@ public function scripts() {
 			$version,
 			true
 		);
+	}
+
+	/**
+	 * Load scripts and styles.
+	 */
+	public function scripts() {
+		// If page is not supported, bail.
+		if ( ! $this->express_checkout_helper->is_page_supported() ) {
+			return;
+		}
+
+		if ( ! $this->express_checkout_helper->should_show_express_checkout_button() ) {
+			return;
+		}
+
+		// Register the script if not already registered
+		if ( ! wp_script_is( 'wc_stripe_express_checkout', 'registered' ) ) {
+			$this->register_express_checkout_script();
+		}
+
+		$asset_path = WC_STRIPE_PLUGIN_PATH . '/build/express-checkout.asset.php';
+		$version    = WC_STRIPE_VERSION;
+		if ( file_exists( $asset_path ) ) {
```

Comment:
We could think of creating a new private method to handle this logic, since it is very similar to the use above.

Reply:
@wjrosa  Thanks for the feedback! I've refactored the asset loading logic into a private `get_asset_data()` method that both `register_express_checkout_script()` and `scripts()` now use.